tensorflow/core/kernels/data/dataset_test_base.cc

```diff
@@ -698,5 +678,100 @@ Status DatasetOpsTestBase::CheckIteratorSaveAndRestore(
   return Status::OK();
 }
 
+Status DatasetOpsTestBaseV2::Initialize(DatasetParams& dataset_params) {
```

Comment:
While we're refactoring the tests, why do we have `DatasetOpsTestBase` and `DatasetOpsTestBaseV2`? Can you add comments that explain the distinction?

Reply:
Before this PR, `DatasetOpsTestBaseV2` will break the existing tests based on `DatasetOpsTestBase`. The changes in this PR will enable us to merge `DatasetOpsTestBaseV2` and `DatasetOpsTestBase`. Will do it once this PR is finalized.